payments/views.py

```diff
@@ -0,0 +1,151 @@
+from decimal import Decimal
+import logging
+import paypalrestsdk
+
+from django.contrib import messages
+from django.http.response import Http404
+from django.shortcuts import redirect
+from django.views.generic import TemplateView
+from django.utils.translation import ugettext as _
+
+from konfera.models import Order
+
+from payments import settings
+from payments.utils import _process_payment
+
+
+logger = logging.getLogger(__name__)
+
+
+paypalrestsdk.configure({
+    "mode": settings.PAYPAL_MODE,
+    "client_id": settings.PAYPAL_CLIENT_ID,
+    "client_secret": settings.PAYPAL_CLIENT_SECRET,
+})
+
+
+class PaymentOptions(TemplateView):
+    template_name = 'payments/order_payment.html'
+
+    def get_order(self, uuid):
+        try:
+            return Order.objects.get(uuid=uuid)
+        except (Order.DoesNotExist, ValueError):
+            raise Http404
+
+    def dispatch(self, request, *args, **kwargs):
+
+        order = self.get_order(kwargs['order_uuid'])
+        if order.status == Order.PAID:
+            return redirect('order_details', order_uuid=str(order.uuid))
+
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['order'] = self.get_order(kwargs['order_uuid'])
+
+        return context
+
+
+class PayOrderByPaypal(TemplateView):
+
+    @staticmethod
+    def get_paypal_price(order):
+        """ Calculate the total price for order when paid by paypal """
+        additional_charge = order.left_to_pay * Decimal(settings.PAYPAL_ADDITIONAL_CHARGE) / Decimal('100')
+        return order.left_to_pay + additional_charge
+
+    @staticmethod
+    def get_paypal_url(payment):
+        """ Return the PayPal URL where the user can pay for the order """
+        for link in payment.links:
+            if link.method == 'REDIRECT':
+                return str(link.href)
+
+        raise Exception("REDIRECT url not found when creating payment {id}".format(id=payment.id))
+
+    def pay(self, request, order):
+        """ Create the payment and redirect to PayPal or back to the order with an error message """
+        payment = paypalrestsdk.Payment({
+            "intent": "sale",
+            "payer": {"payment_method": "paypal"},
+            "redirect_urls": {
+                "return_url": request.build_absolute_uri().split('?')[0] + '?status=success',
+                "cancel_url": request.build_absolute_uri().split('?')[0] + '?status=failed',
+            },
+            "transactions": [
+                {
+                    "amount": {
+                        "total": str(PayOrderByPaypal.get_paypal_price(order)),
+                        "currency": settings.PAYPAL_CURRENCY,
+                    },
+                    "description": _("Payment for order with variable symbol: {vs}".format(vs=order.variable_symbol))
```

Comment:
In order we have property event, why dont we add event name (possibly ticket type) here as well? Will be more user friendly if the person is looking at PayPal's history after some time...